core/src/main/java/org/apache/calcite/rex/RexChecker.java

```diff
@@ -160,15 +155,7 @@ public int getFailureCount() {
       ++failCount;
       return litmus.fail(null);
     }
-    final RelDataTypeField typeField = refType.getFieldList().get(index);
```

Comment:
In the process of my use, the verification here is indeed effective. If only the nullability value is different, we can add 'RelOptUtil. eqIgnoreNullable'` to replace it?

Reply:
Ok, I will add this check back.